src/metabase/db/query.clj

```diff
@@ -34,20 +36,37 @@
 
 (set! *warn-on-reflection* true)
 
+(def ^:private sql-dialects
+  {:db2 Dialect/Db2
+   :mariadb Dialect/MariaDb
+   :mysql Dialect/MySql
+   :n1ql Dialect/N1ql
+   :plsql Dialect/PlSql
+   :postgresql Dialect/PostgreSql
+   :redshift Dialect/Redshift
+   :sparksql Dialect/SparkSql
+   :standardsql Dialect/StandardSql
+   :tsql Dialect/TSql})
+
+(def ^:private SQLDialect
+  (into [:enum {:error/message "valid sql dialect"}] (keys sql-dialects)))
+
+(defmulti sql-formatter-dialect
+  "A dialect to use for pretty printing SQL queries. By default uses StandardSQL"
+  {:added "0.48.0" :arglists '([driver])}
+   driver/dispatch-on-initialized-driver
+   :hierarchy #'driver/hierarchy)
+
+(mu/defmethod sql-formatter-dialect :default :- SQLDialect
+  [_driver]
+  :standardsql)
+
 (defn- format-sql*
   "Return a nicely-formatted version of a generic `sql` string.
   Note that it will not play well with Metabase parameters."
-  [^String sql db-type]
+  [^String sql driver]
   (when sql
-    (let [formatter (SqlFormatter/of (case db-type
-                                       :mysql Dialect/MySql
-                                       :postgres Dialect/PostgreSql
-                                       :redshift Dialect/Redshift
-                                       :sparksql Dialect/SparkSql
-                                       :sqlserver Dialect/TSql
-                                       :oracle Dialect/PlSql
-                                       :bigquery-cloud-sdk Dialect/MySql
-                                       Dialect/StandardSql))]
+    (let [formatter (SqlFormatter/of (sql-dialects (sql-formatter-dialect driver)))]
```

Comment:
We could probably leave the existing case switch as-is, calling the multimethod only for the default case. However, it will require importing `(com.github.vertical_blank.sqlformatter.languages Dialect)` in the driver code that requires dialect overrides, and it looks like a local implementation detail plus, generally speaking, it is not very pretty. Additionally, in case of the formatter library change, the drivers will also have to fix the imports, etc. So I decided to write a thin wrapper over it with some type hints (apologies if I use it in the wrong way) as it probably will be more maintainable.